### PR TITLE
Remove checksums

### PR DESCRIPTION
**Pull request by bmxam**
The checksum of meshes generated by `gmsh_jll` depends on the gmsh version and the architecture (some nodes may slightly vary). In this PR, we the remove checks on mesh checksums in favor of a check with reference serialized mesh. We can then apply a tolerance on the array comparison.

Closes #22 , closes #21 , closes #23 

Note : I don't know why the CI is taking so long...

### CODE REVIEW

File: test/test_read.jl [hunk 1]
```diff
@@ -162,4 +162,38 @@
...
+            18,
+            19,
+            20,
+            21,
+            22,
+            23,
+            24,
+            25,
+            26,
+        ]
```
Comment (ghislainb) [inline test/test_read.jl]: `collect(2:26)`

File: test/utils.jl [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+        # Compare c2n
+        if !compare_connectivities(
+            Bcube.connectivities_indices(mesh_a, :c2n),
+            Bcube.connectivities_indices(mesh_b, :c2n),
+        )
+            verbose && println("Comparison of c2n failed")
+            identical = false
+        end
+
```
Comment (ghislainb) [inline test/utils.jl:71]: Are there any tests for boundary condition tags and elements ?

File: test/test_generators.jl [hunk 3]
```diff
@@ -33,30 +44,54 @@
...
 
     # gen_cylinder_shell_mesh
+    # Use `compare_mesh_nodes_cloud_helper` because node ordering (and c2n) varies between machines...
     fname = "gmsh_cylinder_shell_mesh.msh"
     BcubeGmsh.gen_cylinder_shell_mesh(joinpath(tempdir, fname), 30, 10)
-    @test fname2sum[fname] == bytes2hex(open(sha1, joinpath(tempdir, fname)))
+    @test compare_mesh_nodes_cloud_helper(
+        joinpath(tempdir, fname),
+        joinpath(REF_DIR, first(splitext(fname)) * SERIALIZED_EXT),
+        7e-5,
```
Comment (ghislainb) [inline test/test_generators.jl:86]: If I understand, the tolerance is higher due to the curvature of the geometry that introduce high result sensitivity? Would it be possible to return the "error_max" for all cases in the log ?

File: test/utils.jl [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    )
+end
+
+"""
+    compare_meshes(mesh_a::AbstractMesh, mesh_b::AbstractMesh, tol)
+
+Compare two meshes with the given `tol` for floating numbers. Return true if
+the two meshes are identical (with respect to the tol), false otherwise.
+"""
+function compare_meshes(
```
Comment (ghislainb) [inline test/utils.jl:38]: This function and `compare_connectivities`  may be moved in `Bcube` in the future

Comment (bmxam) [reply test/utils.jl:38]: I agree, it was my first intention, but I'd prefer to move it the day we need it for something else (like BcubeCGNS or BcubeVTK)

File: test/test_read.jl (hunk 1 above)
Comment (bmxam) [reply test/test_read.jl]: done

File: test/utils.jl (hunk 2 above)
Comment (bmxam) [reply test/utils.jl:71]: done

File: test/test_generators.jl (hunk 3 above)
Comment (bmxam) [reply test/test_generators.jl:86]: done

**Review by ghislainb — APPROVED** [review]
